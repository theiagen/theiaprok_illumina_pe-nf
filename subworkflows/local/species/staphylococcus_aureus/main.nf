include { SPATYPER         } from '../../../../modules/local/species_typing/staphylococcus/spatyper/main'
include { STAPHOPIASCCMEC } from '../../../../modules/local/species_typing/staphylococcus/staphopiasccmec/main'
include { AGRVATE         } from '../../../../modules/local/species_typing/staphylococcus/agrvate/main'

workflow STAPHYLOCOCCUS_AUREUS_SPECIES_TYPING {

    take:
    ch_staphylococcus // Channel of tuples [meta, assembly, reads, species]

    main:

    ch_versions = Channel.empty()
    ch_spatyper_results = Channel.empty()
    ch_staphopiasccmec_results = Channel.empty()
    ch_agrvate_results = Channel.empty()

    // Extract assembly for all tools
    ch_assembly = ch_staphylococcus.map { meta, assembly, reads, species -> [meta, assembly] }

    SPATYPER (
        ch_assembly,
        params.spatyper_do_enrich ?: false
    )
    ch_spatyper_results = SPATYPER.out.tsv
    ch_versions = ch_versions.mix(SPATYPER.out.versions)
    
    STAPHOPIASCCMEC (
        ch_assembly
    )
    ch_staphopiasccmec_results = STAPHOPIASCCMEC.out.staphopia_results_tsv
    ch_versions = ch_versions.mix(STAPHOPIASCCMEC.out.versions)
    
    AGRVATE (
        ch_assembly
    )
    ch_agrvate_results = AGRVATE.out.agrvate_summary
    ch_versions = ch_versions.mix(AGRVATE.out.versions)

    emit:
    spatyper_results = ch_spatyper_results
    staphopiasccmec_results = ch_staphopiasccmec_results
    agrvate_results = ch_agrvate_results
    
    versions = ch_versions
}