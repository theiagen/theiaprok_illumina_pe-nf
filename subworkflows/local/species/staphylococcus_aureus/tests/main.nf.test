nextflow_workflow {
    name "Test Subworkflow STAPHYLOCOCCUS_AUREUS_SPECIES"
    script "../main.nf"
    workflow "STAPHYLOCOCCUS_AUREUS_SPECIES_TYPING"

    test("staphylococcus_aureus - comprehensive typing") {
        setup {
            run("GUNZIP") {
                script "../../../../../modules/nf-core/gunzip/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [id:'test_staph_aureus', single_end:false],
                        file('https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/839/195/GCF_001839195.1_ASM183919v1/GCF_001839195.1_ASM183919v1_genomic.fna.gz', checkIfExists: true)
                    ])
                    """
                }
            }
        }
        
        when {
            workflow {
                """
                input[0] = GUNZIP.out.gunzip
                    .map { meta, assembly -> 
                        [
                            meta, 
                            assembly, 
                            [],
                            "Staphylococcus aureus"
                        ]
                    }
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.spatyper_results,
                    workflow.out.staphopiasccmec_results,
                    workflow.out.agrvate_results,
                    workflow.out.versions
                ).match() }
            )
        }
    }

    test("staphylococcus_aureus - spatyper with enrichment") {
        setup {
            run("GUNZIP") {
                script "../../../../../modules/nf-core/gunzip/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [id:'test_staph_enriched', single_end:false],
                        file('https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/839/195/GCF_001839195.1_ASM183919v1/GCF_001839195.1_ASM183919v1_genomic.fna.gz', checkIfExists: true)
                    ])
                    """
                }
            }
        }
        
        when {
            params {
                spatyper_do_enrich = true
            }
            workflow {
                """
                input[0] = GUNZIP.out.gunzip
                    .map { meta, assembly -> 
                        [
                            meta, 
                            assembly, 
                            [],
                            "Staphylococcus aureus"
                        ]
                    }
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.spatyper_results,
                    workflow.out.staphopiasccmec_results,
                    workflow.out.agrvate_results,
                    workflow.out.versions
                ).match() }
            )
        }
    }
}