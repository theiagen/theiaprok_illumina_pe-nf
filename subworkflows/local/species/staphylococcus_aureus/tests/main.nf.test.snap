{
    "staphylococcus_aureus - comprehensive typing": {
        "content": [
            [
                [
                    {
                        "id": "test_staph_aureus",
                        "single_end": false
                    },
                    "test_staph_aureus.tsv:md5,3043d68e3a27de197bf0334d58fbd85b"
                ]
            ],
            [
                [
                    {
                        "id": "test_staph_aureus",
                        "single_end": false
                    },
                    "test_staph_aureus.staphopia-sccmec.summary.tsv:md5,fa2f9b997dcb27e4969fe695abc7ad32"
                ]
            ],
            [
                [
                    {
                        "id": "test_staph_aureus",
                        "single_end": false
                    },
                    "test_staph_aureus.agrvate.tsv:md5,6e87b5fc3b52f82f19c47ae8a0af41ba"
                ]
            ],
            [
                "versions.yml:md5,3a12190cba221f3981e468a75492b33f",
                "versions.yml:md5,4d893c7d995bea1572c9b8620c8af535",
                "versions.yml:md5,fa79ee0840e41ba0d5be60ff591652cb"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.5"
        },
        "timestamp": "2025-07-02T21:11:40.115162"
    },
    "staphylococcus_aureus - spatyper with enrichment": {
        "content": [
            [
                [
                    {
                        "id": "test_staph_enriched",
                        "single_end": false
                    },
                    "test_staph_enriched.tsv:md5,a698352823875171696e5e7ed7015c13"
                ]
            ],
            [
                [
                    {
                        "id": "test_staph_enriched",
                        "single_end": false
                    },
                    "test_staph_enriched.staphopia-sccmec.summary.tsv:md5,fa2f9b997dcb27e4969fe695abc7ad32"
                ]
            ],
            [
                [
                    {
                        "id": "test_staph_enriched",
                        "single_end": false
                    },
                    "test_staph_enriched.agrvate.tsv:md5,6e87b5fc3b52f82f19c47ae8a0af41ba"
                ]
            ],
            [
                "versions.yml:md5,3a12190cba221f3981e468a75492b33f",
                "versions.yml:md5,4d893c7d995bea1572c9b8620c8af535",
                "versions.yml:md5,fa79ee0840e41ba0d5be60ff591652cb"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.5"
        },
        "timestamp": "2025-07-02T21:11:57.580372"
    }
}